### PR TITLE
Swift 4.2: Rename Random to SystemRandomNumberGenerator

### PR DESCRIPTION
**Pull request by greg**
Another code-breaking change for Swift 4.2 🙂

Side note: I also had to change the Swift version to 4.2 (from 4.0) in the Xcode project build settings for this to work.

### CODE REVIEW

File: .swift-version [hunk 1]
```diff
@@ -1 +1 @@
-4.1
+4.2
```
Comment (stephencelis) [inline .swift-version]: I think this needs to be locked to a snapshot for now. Can you try changing this line to:

```
swift-4.2-DEVELOPMENT-SNAPSHOT-2018-08-03
```

File: Sources/NonEmpty/NonEmpty+Random.swift [hunk 2]
```diff
@@ -5,7 +5,12 @@ extension NonEmpty {
   }
 
   public func randomElement() -> Element {
+    #if swift(>=4.2)
+    var generator = SystemRandomNumberGenerator()
```
Comment (stephencelis) [inline Sources/NonEmpty/NonEmpty+Random.swift]: Don't have time to check right now, but did this get back-ported to 4.1.x compatibility mode in the latest Xcode?

File: .swift-version (hunk 1 above)
Comment (greg) [reply .swift-version]: Nope, still fails:

> We don't have build instructions for 4.2-DEVELOPMENT-SNAPSHOT-2018-08-03.

**Comment by greg** [comment]
After changing `.swift-version` to `swift-4.2-DEVELOPMENT-SNAPSHOT-2018-08-03-a`, and **removing** the 4.2 check, it got as far as compiling, but something is still broken...

I'll investigate this further in a bit, sorry for abusing the CI 😛

**Comment by greg** [comment]
So it turns out the crash was unrelated to my changes, but something went wrong with the Swift 4.2 compiler and `RandomAccessCollection`...

The build is passing because I commented out the extension that adds that conformance, so it's probably not a good idea to merge this PR until I figure out a fix.

(we should also probably report this compiler bug)

**Comment by stephencelis** [comment]
Hey @greg, thanks for your patience! I'm gonna merge this as is and we'll get around to fixing the CI situation soon. Thanks again for contributing!

**Pull request merged**